R/learner_stats_regr_glm.R

```diff
@@ -0,0 +1,116 @@
+#' @title Regression Generalized Linear Model Regression Learner
+#' @author salauer
+#' @name mlr_learners_regr.glm
+#'
+#' @template class_learner
+#' @templateVar id regr.glm
+#' @templateVar caller glm
+#'
+#'
+#' @template seealso_learner
+#' @template example
+#'
+#' @section Custom mlr3 defaults:
+#' - `type`
+#'   - Actual default: "link"
+#'   - Adjusted default: "response"
+#'   - Reason for change: Users rarely want predictions on the link scale
+#'   (for instance, on the log scale when using the Poisson distribution).
+#'   It would be much more reasonable for the default to be "response",
+#'   which puts predictions on the scale of the observed data,
+#'   and allow the user to change the type to "link" if they so desire.
+#' @export
+LearnerRegrGlm = R6Class("LearnerRegrGlm",
+  inherit = LearnerRegr,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      ps = ParamSet$new(list(
+        ParamLgl$new("singular.ok", default = TRUE, tags = "train"),
+        ParamLgl$new("x", default = FALSE, tags = "train"),
+        ParamLgl$new("y", default = TRUE, tags = "train"),
+        ParamLgl$new("model", default = TRUE, tags = "train"),
+        ParamUty$new("etastart", tags = "train"),
+        ParamUty$new("mustart", tags = "train"),
+        ParamUty$new("start", default = NULL, tags = "train"),
+        ParamUty$new("offset", tags = "train"),
+        ParamFct$new("family",
+          default = "gaussian",
+          levels = c("gaussian", "poisson", "quasipoisson", "Gamma", "inverse.gaussian"),
+          tags = "train"),
+        ParamDbl$new("epsilon", default = 1e-8, tags = c("train", "control")),
+        ParamDbl$new("maxit", default = 25, tags = c("train", "control")),
+        ParamLgl$new("trace", default = FALSE, tags = c("train", "control")),
+        ParamUty$new("dispersion", default = NULL, tags = "predict"),
+        ParamFct$new("type",
+          default = "link",
+          levels = c("response", "link", "terms"),
+          tags = "predict")
+      ))
```

Comment:
Minor point but please remove line breaks if not needed

Reply:
`lintr` was angry at long lines. I reduced linebreaks to a minimum.